dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/notification/DefaultProgramNotificationService.java

```diff
@@ -451,7 +451,7 @@ private ProgramMessage createProgramMessage( ProgramInstance programInstance, Pr
 
         if ( recipientType == ProgramNotificationRecipient.USER_GROUP )
         {
-            recipients = template.getRecipientUserGroup().getMembers();
+            recipients = Optional.ofNullable( template.getRecipientUserGroup().getMembers() ).orElse( recipients );
```

Comment:
This would be safer, feel free to apply if you feel like:

```suggestion
            recipients = Optional.ofNullable( template )
                    .map( ProgramNotificationTemplate::getRecipientUserGroup )
                    .map( UserGroup::getMembers )
                    .orElse( recipients );
```